rayon-core/src/registry.rs

```diff
@@ -857,19 +865,37 @@ impl WorkerThread {
     }
 
     pub(super) fn yield_now(&self) -> Yield {
+        let yielding = &self.registry.thread_infos[self.index].is_yielding;
```

Comment:
Seems this could just be a `Cell<bool>` in `WorkerThread` (since that's where the thread-local stuff generally lives) instead of living on the registry?